src/admin_extra/service_admin_extra_migration.erl

```diff
@@ -0,0 +1,788 @@
+%% @end
+%% -------------------------------------------------------------------
+-spec try_migrate(Act :: binary() | {binary(), atom()}) -> Result :: tuple().
+
+try_migrate(<<"all">>) ->
+  Acts = [<<"pubsub_nodes">>, <<"pubsub_affiliations">>, <<"pubsub_subscriptions">>, <<"pubsub_items">>,
+          <<"vcard">>, <<"vcard_search">>, <<"users">>, <<"event_pusher_push_subscription">>,
+          <<"rosterusers">>, <<"roster_version">>, <<"rostergroups">>, <<"last">>, <<"private_storage">>,
+          <<"offline_message">>, <<"muc_light_rooms">>],
+  _ = [try_migrate(Act) || Act <- Acts],
+  {ok, io_lib:format("Completed the migration of the tables: ~p", [Acts])};
+
+try_migrate(Act) when is_binary(Act) ->
+  case Act of
+    <<"pubsub_nodes">> ->
+      try_migrate({Act, pubsub_node});
+    Act when Act =:= <<"pubsub_affiliations">>; Act =:= <<"pubsub_subscriptions">> ->
+      try_migrate({Act, pubsub_state});
+    <<"pubsub_items">> ->
+      try_migrate({Act, pubsub_item});
+    <<"vcard">> ->
+      try_migrate({Act, vcard});
+    <<"vcard_search">> ->
+      try_migrate({Act, vcard_search});
+    <<"users">> ->
+      try_migrate({Act, passwd});
+    <<"event_pusher_push_subscription">> ->
+      try_migrate({Act, push_subscription});
+    Act when Act =:= <<"rosterusers">>; Act =:= <<"rostergroups">> ->
+      try_migrate({Act, roster});
+    <<"roster_version">> ->
+      try_migrate({Act, roster_version});
+    <<"last">> ->
+      try_migrate({Act, last_activity});
+    <<"private_storage">> ->
+      try_migrate({Act, private_storage});
+    <<"offline_message">> ->
+      try_migrate({Act, offline_msg});
+    <<"muc_light_rooms">> ->
+      try_migrate({Act, muc_light_room});
+    Act ->
+      {error, io_lib:format("Migrate dont support act: ~p", [binary_to_list(Act)])}
+  end;
+
+try_migrate({Act, Table}) when is_atom(Table) ->
+  case get_all_data(Table) of
+    {ok, []} ->
+      {error, io_lib:format("The '~p' table in Mnesia is empty", [Act])};
+    {ok, Data} ->
+      ?MODULE:(list_to_existing_atom("migrate_" ++ binary_to_list(Act)))(Data);
+    Error ->
+      Error
+  end.
+
+%% -------------------------------------------------------------------
+%% @private
+%% @doc
+%% Helper for migrate "pubsub_node" from Mnesia to SQL DB
+%% @end
+%% -------------------------------------------------------------------
+-spec migrate_pubsub_nodes(Data :: list()) -> Result :: tuple().
+
+migrate_pubsub_nodes([]) ->
+  {ok, "Completed the migration of the table 'pubsub_nodes'"};
+
+migrate_pubsub_nodes([H|T]) ->
+  {Host, ID} = H#pubsub_node.nodeid,
+  Owners = jiffy:encode([jid:to_binary(O) || O <- H#pubsub_node.owners]),
+  Ops = jiffy:encode({H#pubsub_node.options}),
+  Cols = ["nidx", "p_key", "name", "type", "owners", "options"],
+  Vals = [H#pubsub_node.id, Host, ID, H#pubsub_node.type, Owners, Ops],
+  Q = ["INSERT INTO pubsub_nodes ", expand_sql_vals(Cols, Vals), ";"],
+  case mongoose_rdbms:sql_query(?HOST, Q) of
```

Comment:
I think here we could use the the pubsub `Host` from `.nodeid` to get the main XMPP host by calling function  `mongoose_subhost:get_host(Host)`